packages/fxa-event-broker/src/pubsub-proxy/pubsub-proxy.controller.ts

```diff
@@ -32,54 +34,93 @@ export class PubsubProxyController {
   @UseGuards(GoogleJwtAuthGuard)
   @Post(':clientId')
   async proxy(
-    @Res() res: Response,
     @Body() body: Request['body'],
     @Param('clientId') clientId: string
   ): Promise<void> {
+    const webhookEndpoint = this.lookupWebhookEndpoint(clientId);
+    const message = this.extractMessage(body.message.data);
+    const queueDelay = Date.now() - message.timestamp;
+    const jwtPayload = await this.generateSET(message, clientId);
+    const { status, data } = await this.proxyMessage({
+      clientId,
+      jwtPayload,
+      webhookEndpoint,
+      message,
+    });
+
+    this.log.debug('proxyDeliverSuccess', {
+      statusCode: status,
+    });
+
+    // Timing for how long the message was in the queue, reported here
+    // only if the message was sent successfully to avoid repeat metrics
+    // for failures.
+    this.metrics.timing(`proxy.queueDelay`, queueDelay);
+
+    throw new HttpException(data, status);
```

Comment:
Without using a Response object, a status code can only be set with a decorator, or on a `HttpException`.

Reply:
Maybe I'm just not that familiar with NestJS yet, but this seems like R- material to me.

If you need to use a decorator to set a status code, why not do that?

Using exception handling for ordinary flow control is an antipattern and throws off JIT execution speed

Reply:
Using a StatusCode decorator means the status code is defined in advance, you cannot dynamically change the statuscode then without using an `HttpException` as done here. In this case, the goal of this route is to proxy the status/response from the Relying Party to Google PubSub so that if its a 503/etc it is automatically retried by Google.